### PR TITLE
Sanitize WS api/issues/*

### PR DESCRIPTION
**Pull request by teryk**

### CODE REVIEW

File: server/sonar-server/src/main/java/org/sonar/server/issue/ws/SearchAction.java [hunk 1]
```diff
@@ -100,6 +103,7 @@
 public class SearchAction implements IssuesWsAction {
 
   private static final String INTERNAL_PARAMETER_DISCLAIMER = "This parameter is mostly used by the Issues page, please prefer usage of the componentKeys parameter. ";
+  private static final Set<String> IGNORED_FACETS = newHashSet(PARAM_PLANNED, DEPRECATED_PARAM_ACTION_PLANS, PARAM_REPORTERS);
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/issue/ws/SearchAction.java]: MINOR Remove this use of "PARAM_REPORTERS"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: server/sonar-server/src/main/java/org/sonar/server/issue/ws/SearchAction.java [hunk 2]
```diff
@@ -176,20 +177,8 @@ public void define(WebService.NewController controller) {
     action.createParam(PARAM_TYPES)
       .setDescription("Comma-separated list of types.")
       .setSince("5.5")
-      .setPossibleValues(RuleType.values())
+      .setPossibleValues((Object[]) RuleType.values())
       .setExampleValue(format("%s,%s", RuleType.CODE_SMELL, RuleType.BUG));
-    action.createParam(DEPRECATED_PARAM_ACTION_PLANS)
```
Comment (julienlancelot) [inline server/sonar-server/src/main/java/org/sonar/server/issue/ws/SearchAction.java]: Removing these parameters should also remove its usage in IssueService, IssueIndex, IssuesWsParameters, SearchAction. No need to have facets on them

**Comment by julienlancelot** [comment]
Am I missing something or there's nothing related to https://jira.sonarsource.com/browse/SONAR-9308 ?

File: sonar-ws/src/main/protobuf/ws-issues.proto [hunk 3]
```diff
@@ -120,10 +120,9 @@ message Flow {
 }
 
 message Location {
-  optional string componentId = 1;
```
Comment (julienlancelot) [inline sonar-ws/src/main/protobuf/ws-issues.proto]: Please keep it as unusedXXX

**Pull request closed without merging**